batch-add-recording-relationships.user.js

```diff
@@ -1079,30 +1088,39 @@ function batch_recording_rels() {
 
             $title_cell.css('color', 'LightSlateGray').find('a').css('color', 'LightSlateGray');
 
-            create_new_work(title, function (data) {
-                let work = data.match(/\/work\/([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})/);
-                let promise = relate_to_work($row, work[1], title, '', true, true);
-
-                if (--total_rows === 0) {
-                    promise.done(function () {
-                        flush_work_cache();
-                        ws_requests.stopped = false;
-                        ws_requests.start_queue();
-                        $button.attr('disabled', false).css('color', '#565656');
-                    });
-                }
-            });
+            create_new_work(
+                title,
+                function (data) {
+                    let work = data.match(/\/work\/([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})/);
+                    let promise = relate_to_work($row, work[1], title, '', true, true, 'new work using recording title');
+
+                    if (--total_rows === 0) {
+                        promise.done(function () {
+                            flush_work_cache();
+                            ws_requests.stopped = false;
+                            ws_requests.start_queue();
+                            $button.attr('disabled', false).css('color', '#565656');
+                        });
+                    }
+                },
+                'new work using recording title'
+            );
         });
     }
 
-    function create_new_work(title, callback) {
+    function create_new_work(title, callback, edit_mode) {
         function post_edit() {
             let data = `edit-work.name=${title}`;
             Object.entries($work_options).forEach(function ([kind, $obj]) {
                 if ($obj.val()) {
                     data += `&edit-work.${kind}_id=${$obj.val()}`;
                 }
             });
+            data += `&edit-work.edit_note=${document.getElementById('bpr-edit-note').value.trim()}${edit_note_signature.replace(
+                /%edit-mode%/g,
+                edit_mode
+            )}`;
```

Comment:
Yeah I totally don't understand all those $ library things.
I will check that!
Thanks!

Reply:
Maybe we should try to convert work create to POST request, otherwise GET could block huge edit notes...

But I think there is an issue with POST /work/create, as seen in my JASRAC work importer, or something...

I have used `encodeURIComponent` in my new commit (cdc1e8299678cb154df22cb6fb9619232743cb70).